pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -169,6 +214,9 @@ def detach_data(model: "pl.LightningModule") -> None:
             if isinstance(loader, _PatchDataLoader):
                 loader.unpatch(model)
 
+    def teardown(self) -> None:
+        [attr.teardown() for attr in vars(self).values() if isinstance(attr, AbstractDataFetcher)]
```

Comment:
```suggestion
        for attr in vars(self).values():
            if isinstance(attr, AbstractDataFetcher):
                attr.teardown()
```
No need to create a list here :)

Reply:
Done !